src/main/java/io/github/christophermanahan/carnitas/HTTPServer.java

```diff
@@ -1,20 +1,20 @@
 package io.github.christophermanahan.carnitas;
 
-public class HTTPServer {
-
+class HTTPServer {
     private final Listener listener;
     private final Logger errorLogger;
     private Connection connection;
 
-    public HTTPServer(Listener listener, Logger errorLogger) {
+    HTTPServer(Listener listener, Logger errorLogger) {
         this.listener = listener;
         this.errorLogger = errorLogger;
     }
 
-    public void run() {
+    void run() {
         try {
             connect();
-            serveUntilDisconnect();
+            serve();
+            close();
```

Comment:
What would happen if an exception occurred before we got to this line?

Reply:
The connection would not close. But close also throws an exception. Should I have another try catch after this? Or should I place close in the catch and not worry about its exception? This is making me think that close should potentially not throw an exception.